modules/admindebug.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2004-2018 ZNC, see the NOTICE file for details.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <znc/FileUtils.h>
+#include <znc/Server.h>
+#include <znc/IRCNetwork.h>
+#include <znc/User.h>
+#include <znc/ZNCDebug.h>
+
+class CAdminDebugMod : public CModule {
+  public:
+    MODCONSTRUCTOR(CAdminDebugMod) {
+        AddHelpCommand();
+        AddCommand("Enable", "", t_d("Enable Debug Mode"),
+                   [=](const CString& sLine) { CommandEnable(sLine); });
+        AddCommand("Disable", "", t_d("Disable Debug Mode"),
+                   [=](const CString& sLine) { CommandDisable(sLine); });
+        AddCommand("Status", "", t_d("Show the Debug Mode status"),
+                   [=](const CString& sLine) { CommandStatus(sLine); });
+    }
+
+    bool OnLoad(const CString& sArgs, CString& sMessage) override {
+        if (GetUser() != nullptr) {
+            return GetUser()->IsAdmin();
+        }
+
+        return true;
+    }
+
+    void CommandEnable(const CString& sCommand) {
+        if (GetUser()->IsAdmin() == false) {
+            PutModule(t_s("Access denied!"));
+            return;
+        }
+
+        ToggleDebug(true, GetUser()->GetNick());
+    }
+
+    void CommandDisable(const CString& sCommand) {
+        if (GetUser()->IsAdmin() == false) {
+            PutModule(t_s("Access denied!"));
+            return;
+        }
+
+        ToggleDebug(false, GetNV("enabled_by"));
+    }
+
+    bool ToggleDebug(bool bEnable, CString sEnabledBy) {
+        bool bValue = CDebug::Debug();
+
+        if (bEnable == bValue) {
+            PutModule(t_s("Already " + CString(bEnable ? "enabled" : "disabled") + "."));

```

Comment:
This isn't possible to translate, even though it does say "t_s".
```c++
if (...) {
  PutModule(t_s("enabled..."));
} else {
  PutModule(t_s("disabled..."));
}
```

Reply:
The previous code before I test it was `t_s("Already %s.", CString(...))`, but t_s doesn't support printf syntax. I will change this.

Reply:
With printf-like syntax, it would be `t_f("Already {1}.")(CDebug::Debug())`, but that's harder to translate - if you provide a full sentence, the translation can be more natural, e.g. not in every language "does" and "does not" can be represented using the same structure.